### DISCUSSION TITLE
What is the canonical way to handle advance payments in Kimai?

### DISCUSSION THREAD

**Discussion by davidpfahler** [kimai/kimai#4155] [General]
### Describe the problem

As an attorney, I routinely ask my clients for an advance payment which usually covers only a small portion of the final fee once the case (i.e. project) is done. I have three unanswered questions in this regard:

1. How should I generate the invoice for the advance payment? I could probably just add a time entry with a fixed price for the "advance payment" activity, but that seems like a hack, and I know @kevinpapst is not a fan of hacks.
2. I invoice my clients every month so there are no surprises regarding the cost. In the first month I may or may not work on the project so much that the billable amount exceeds the advance payment. In case the advance payment is exceeded, I would like to create an invoice that shows all time entries, the total duration and total amount, but subtracts the advance payment from the total amount. As @kevinpapst has closed #3008, I am not sure how to accomplish this.
3. How can I ensure that the advance payment is only deducted once and not on any subsequent invoice?

### Describe the solution you'd like

As @kevinpapst seem to be opposed to all suggestions made in #3008 I am not sure how to solve this. I would really appreciate any feedback or ideas.

**Comment by kevinpapst** [comment]
Use the expenses bundle and create a negative amount entry for that client with the negative amount.
It will be deducted from the first invoice (unless marked exported). 
That is the "official" way.

Kimai will not allow to create invoices with an amount smaller or equal to zero.
Therefor 2 might not be able unless your work exceeds the pre-payment. But an invoice makes no sense anyway without an invoicable amount. You can probably still use the export to create an overview for the client.

**Reply by davidpfahler** [reply]
Thanks for the quick response! If there is no way to create invoice with negative amounts, how can I handle cancellation of invoice („Stornierung“ in German). According to my tax advisor, it is required to create a negative invoice (either negative amount of negative total) for the canceled invoice.
